astropy/coordinates/representation/base.py

```diff
@@ -1323,6 +1336,8 @@ def __init_subclass__(cls, **kwargs):
         For these, the components are those of the base representation prefixed
         by 'd_', and the class is `~astropy.units.Quantity`.
         """
+        super().__init_subclass__(**kwargs)  # sets `cls.name`
```

Comment:
As above, maybe after the sanity checks.

Reply:
Resolving, from https://github.com/astropy/astropy/pull/17503#discussion_r1871782217